cluster-backup-tool.md

```diff
@@ -0,0 +1,144 @@
+Cluster Backup Tool
+====================
+# Requirements
+## Hard
+- Ability to:
+    - Backup data to a file(s) at a given point in time
```

Comment:
any idea how to achieve this?

Reply:
I'm not sure I understand the question. This whole document is essentially a proposal on how to do ^.

Reply:
so, transactional caches cross my mind, more concretely, how to achieve a valid snapshot.
Since tx involves multiple [primary] owners, iterating over the keys may not be a good idea. And starting a cluster from backup with some partial transactions committed may not be a good idea.

Reply:
So the initial scope of this tool is to create a backup from an "offline" cluster, i.e. one that is not currently serving requests, so in this case there won't be any pending transactions. If one or more transactions are pending on a cache, we could potentially wait for them to complete before proceeding.

Eventually we will want "online" backups of a live cluster, which will definitely have to take into account transactional boundarys and defining what a valid snapshot is. This is non-trivial and realistically will probably be best suited to a breakout session at our next F2F (whenever that will be). I make reference to this in the `Optional/Future` section here: https://github.com/infinispan/infinispan-designs/pull/17/files#r418168778